scRNA-seq/02-normalizing_scRNA-seq.Rmd

```diff
@@ -67,185 +72,355 @@ sc_metadata <- readr::read_tsv(metadata_file)
+We've plotted the counts data for you. Knowing that we want the same graph, but different data, use the above template to plot the normalized data. 
+Feel free to customize the plot with a different theme or color scheme!
 
 Let's plot the `norm_pca_scores` data:
 
-```{r Plotting norms pca, live = TRUE}
+```{r norm_pca_plot, live = TRUE}
 ggplot(norm_pca_scores, aes(x = PC1, y = PC2, color = cell_type)) +
   geom_point() + 
-  ggtitle("Normalized log counts PCA Scores") + 
+  ggtitle("Normalized log counts PCA scores") + 
+  colorblindr::scale_color_OkabeIto()
+```
+
+Do you see an effect from the normalization in the comparison between these plots?
+
+## Modeling variance
+
+The variation in gene expression we see among cells comes from a combination of variation due to technical effects and the biology we really care about. 
+In order to account for this we could just take the largest variance genes, but if we want to be a bit more careful about it, we can model the variance in expression of each gene as a function of mean expression for that gene.
+This is useful because we generally expect the variance to increase as mean expression increases, even if there is no biological signal in the expression variation.
+
+We will do this modeling of variance by expressionwith the `scran::modelGeneVar()` function, saving the results to a new variable. 
+
+```{r model_variance}
+gbm_variance <- scran::modelGeneVar(gbm_sce)
+```
+
+Now lets plot the relationship between gene expression and variance we were discussing.
+Here we will also add the fitting curve that `scran::modelGeneVar()` created, which is stored as function in the  `$trend` slot of the `gbm_variance` object.
+We can add a function like that curveto a `ggplot` with a `stat_function` layer.
+
+```{r plot_variance}
+ggplot(as.data.frame(gbm_variance), aes(x = mean, y = total)) +
+  geom_point(alpha = 0.05) +
+  stat_function(fun = metadata(gbm_variance)$trend, color = "blue") + 
+  xlab("Mean log-expression") + 
+  ylab("Variance")
+
+```
+
+Now we can use `scran::getTopHVGs()` to select the genes that have the most biological variation (according to the model) and calculate PCA scores using only those genes. 
+Here we are picking the 500 top genes.
+
+```{r get_highvar}
+highvar_genes <- scran::getTopHVGs(gbm_variance, n = 500)
+highvar_pca <- scater::calculatePCA(gbm_sce, subset_row = highvar_genes)
+```
+
+```{r highvar_df}
+highvar_pca_scores <- data.frame(highvar_pca,
+                                 geo_accession = rownames(highvar_pca),
+                                 cell_type = gbm_sce$cell_type)
+```
+
+```{r plot_highvar}
+ggplot(highvar_pca_scores, aes(x = PC1, y = PC2, color = cell_type)) +
+  geom_point() + 
+  ggtitle("High variance gene PCA scores") + 
+  colorblindr::scale_color_OkabeIto()
+```
+
+Now what we didn't tell you before, but will tell you now, is that the default for `calculatePCA()` is to use the top 500 highest variance genes. 
```

Comment:
Not clear to me from my quick read of documentation if `calculatePCA()` includes the `modelGeneVar()` `getTopHVGs()` steps, but I strongly suspect you've read more about this than me and also that design choice seems super reasonable.